### PR TITLE
OGM-983 Create integration with Apache Ignite (and GridGain)

### PR DESCRIPTION
**Pull request by schernolyas**
the commit contains :
1) automatic schema generation
2) support read-thought and write-thought (removed)
3) integration for Apache Ignite transactions
4) support Ignite Native Persistence
5) migration to Apache Ignite 2.1

"support read-thought and write-thought" removed from the PR because:
1) it is removed from my project (migration to Ignite Native Persistence)
2) it is not ready for wide using (many problems with associations)

### CODE REVIEW

**Comment by DavideD** [comment]
I'm having trouble building this branch after a rebase. I'll continue tomorrow after I've set up a job on CI

File: ignite/src/main/java/org/hibernate/ogm/datastore/ignite/transaction/impl/IgniteJtaTransactionCoordinator.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	private void join() {
+		Ignite ignite = datastoreProvider.getCacheManager();
+		IgniteTransactions igniteTransactions = ignite.transactions();
+
+		/*
+		 * if ( currentOrientDBTransaction == null && delegate.isActive() ) { log.debugf(
+		 * "begin transaction for database %s", database.getName() ); database.begin(); currentOrientDBTransaction =
+		 * database.getTransaction(); delegate.getLocalSynchronizations().registerSynchronization( new
+		 * OrientDBSynchronization() ); }
+		 */
```
[the remark this answers is not in the document]
Comment (schernolyas) [reply ignite/src/main/java/org/hibernate/ogm/datastore/ignite/transaction/impl/IgniteJtaTransactionCoordinator.java]: fixed!

File: ignite/src/main/java/org/hibernate/ogm/datastore/ignite/transaction/impl/IgniteJtaTransactionCoordinator.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	 * currentOrientDBTransaction = null; } } private void failure() { if ( currentOrientDBTransaction != null ) {
+	 * log.debugf( "rollback  transaction N %d for database %s", currentOrientDBTransaction.getId(),
+	 * currentOrientDBTransaction.getDatabase().getName() ); currentOrientDBTransaction.rollback();
+	 * currentOrientDBTransaction = null; } } private class OrientDBSynchronization implements Synchronization {
+	 * @Override public void beforeCompletion() { TransactionStatus status =
+	 * delegate.getTransactionDriverControl().getStatus(); if ( status == TransactionStatus.MARKED_ROLLBACK ) {
+	 * failure(); } else { success(); } }
+	 * @Override public void afterCompletion(int status) { if ( currentOrientDBTransaction != null ) { if ( status !=
+	 * Status.STATUS_COMMITTED ) { failure(); } else { success(); } } } }
+	 */
```
[the remark this answers is not in the document]
Comment (schernolyas) [reply ignite/src/main/java/org/hibernate/ogm/datastore/ignite/transaction/impl/IgniteJtaTransactionCoordinator.java]: fixed

File: distribution/pom.xml [hunk 3]
```diff
@@ -10,7 +10,7 @@
     <parent>
         <groupId>org.hibernate.ogm</groupId>
         <artifactId>hibernate-ogm-ignite-parent</artifactId>
-        <version>5.2.0-SNAPSHOT</version>
+        <version>5.2.0.Alpha1</version>
```
[the remark this answers is not in the document]
Comment (schernolyas) [reply distribution/pom.xml]: fixed

**Comment by DavideD** [comment]
I'm going to close this PR and send a new one to the new official repository, but I will  need your help to understand how to make the tests work

**Pull request closed without merging**